app/api/api.ts

```diff
@@ -1,27 +1,36 @@
 import axios from 'axios';
+import urljoin from 'url-join';
 import {
   Transaction,
   TransactionResults,
   MempoolTransaction,
-  AccountBalance,
+  AddressBalanceResponse,
 } from '@blockstack/stacks-blockchain-api-types';
 
-const api = 'https://sidecar.staging.blockstack.xyz/sidecar';
+export class Api {
+  constructor(public baseUrl: string) {}
 
-async function getAddressBalance(address: string) {
-  return axios.get<AccountBalance>(api + `/v1/address/${address}/balances`);
-}
+  async getAddressBalance(address: string) {
+    return axios.get<AddressBalanceResponse>(
+      urljoin(this.baseUrl, `v1/address/${address}/balances`)
```

Comment:
I didn't really wanna use a library here, but new `new URL` doesn't really do path joining. Given users can add their own urls, it may, or may not, end with a trailing `/`.

Maybe we could strip it when they add it, to be sure? But in this case I just used a lib that'll concat urls and handle any cases that'll avoid `v1/extended//status`